src/wasm-lib/kcl/src/ast/types.rs

```diff
@@ -2834,29 +2834,74 @@ impl MemberExpression {
     }
 
     pub fn get_result(&self, memory: &mut ProgramMemory) -> Result<MemoryItem, KclError> {
-        let property_name = match &self.property {
-            LiteralIdentifier::Identifier(identifier) => identifier.name.to_string(),
+        #[derive(Debug)]
+        enum Property {
+            Number(usize),
+            String(String),
+        }
+
+        impl Property {
+            fn type_name(&self) -> &'static str {
+                match self {
+                    Property::Number(_) => "number",
+                    Property::String(_) => "string",
+                }
+            }
+        }
+
+        let property_src: SourceRange = self.property.clone().into();
+        let property_sr = vec![property_src.clone()];
+
+        let property: Property = match self.property.clone() {
+            LiteralIdentifier::Identifier(identifier) => {
+                let prop = memory.get(&identifier.name, property_src)?;
```

Comment:
From looking at `recast()`, I think we actually need to condition on `self.computed` and if it's false, use `identifier.name.to_string()` (the old behavior) in that case.

I wouldn't have expected that both `obj[x]` and `obj.x` parse to the same AST node type with a boolean flag.

Reply:
The issue is that we cannot currently parse a more complicated indexing expression like this:

```javascript
obj[i + 1]
```